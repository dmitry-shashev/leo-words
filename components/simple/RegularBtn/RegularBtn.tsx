import { FC } from 'react'

interface Props {
  label: string
  color: 'blue' | 'green' | 'red'
  onClick: () => void
  ariaLabel?: string
}

export const RegularBtn: FC<Props> = ({ label, color, onClick, ariaLabel }) => {
  let bgColor: string
  switch (color) {
    case 'blue':
      bgColor = 'bg-blue-500'
      break
    case 'red':
      bgColor = 'bg-red-700'
      break
    case 'green':
      bgColor = 'bg-green-700'
      break
  }

  return (
    <button
      aria-label={ariaLabel ?? label}
      onClick={onClick}
      className={`${bgColor} text-3xl font-semibold text-white h-16 rounded grow`}
      type="button"
    >
      {label}
    </button>
  )
}
