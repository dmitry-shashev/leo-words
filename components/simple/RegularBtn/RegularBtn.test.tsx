import { expect, test, vi } from 'vitest'
import { render, fireEvent } from '@testing-library/react'
import { RegularBtn } from '@/components/simple/RegularBtn/RegularBtn'

test('RegularBrn', () => {
  const onClick = vi.fn()
  const label = 'Some btn'
  const { asFragment, getByRole } = render(
    <RegularBtn label={label} onClick={onClick} color="green" />
  )

  expect(asFragment()).toMatchSnapshot()

  expect(onClick).toBeCalledTimes(0)
  fireEvent.click(getByRole('button'))
  expect(onClick).toBeCalledTimes(1)
})

test('RegularBrn - medium', () => {
  const onClick = vi.fn()
  const label = 'Some medium btn'
  const { asFragment } = render(
    <RegularBtn label={label} onClick={onClick} color="green" size="medium" />
  )

  expect(asFragment()).toMatchSnapshot()
})
