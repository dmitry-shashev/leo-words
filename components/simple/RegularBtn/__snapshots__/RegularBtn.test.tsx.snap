// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegularBrn 1`] = `
<DocumentFragment>
  <button
    class="bg-green-700 text-3xl font-semibold text-white h-16 rounded grow"
    type="button"
  >
    Some btn
  </button>
</DocumentFragment>
`;
